specification/sections/02-providers.md

```diff
@@ -150,3 +150,70 @@ class MyProvider implements Provider {
 #### Requirement 2.3.3
 
 > In cases of abnormal execution, the `resolution details` structure's `error message` field **MAY** contain a string containing additional detail about the nature of the error.
+
+### 2.4 Initialization
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Requirement 2.4.1
+
+> The `provider` interface **MUST** define an `initialize` function which accepts the global `evaluation context` as a argument, which can be optionally implemented to perform initialization logic relevant to the provider.
+
+```
+class MyProvider implements Provider {
+  //...
+
+  void initialize(EvaluationContext initialContext) {
+    /*
+      A hypothetical initialization function: connect to a flag database, then pass a callback which fires PROVIDER_READY
+    */
+    this.connectToDatabase(() => this.emitEvent(ProviderEvents.READY));
+  }
+
+  //...
+}
+```
+
+### 2.5. Provider context reconciliation
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+Client-focused providers may need a mechanism to understand when their cache of evaluated flags must be invalidated or updated. An `on-context-set` handler can be defined which performs whatever operations are needed to reconcile the evaluated flags with the new context.
+
+#### Requirement 2.5.1
+
+> The provider interface **MUST** define an `on context set` handler, which takes a argument for the previous context, and the newly set context, and which can be optionally implemented to reconcile any stored state pertaining to the global evaluation context.
+
+Especially in single-context implementations, providers and underlying SDKs my maintain a cache of evaluated flags for a particular context.
+The `on context set` handler provides a mechanism to update this state, often by re-evaluating flags in bulk with respect to the new context.
+
+```
+  // run the myOnReadyHandler function when the 
```

Comment:
when the ...

Reply:
This isn't even the correct example, I missed this one!